vochain/ist/store.go

```diff
@@ -0,0 +1,52 @@
+package ist
+
+import (
+	"encoding/binary"
+
+	"go.vocdoni.io/dvote/crypto/ethereum"
+)
+
+func (c *Controller) storeToNoState(key, value []byte) error {
+	tx, err := c.st.Store.BeginTx()
+	if err != nil {
+		return err
+	}
+	defer tx.Discard()
+	if err := tx.NoState().Set(key, value); err != nil {
+		return err
+	}
+	return tx.SaveWithoutCommit()
+}
+
+func (c *Controller) deleteFromNoState(key []byte) error {
+	tx, err := c.st.Store.BeginTx()
+	if err != nil {
+		return err
+	}
+	defer tx.Discard()
+	if err := tx.NoState().Delete(key); err != nil {
+		return err
+	}
+	return tx.SaveWithoutCommit()
+}
+
+func (c *Controller) retrieveFromNoState(key []byte) ([]byte, error) {
+	tx, err := c.st.Store.BeginTx()
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Discard()
+	return tx.NoState().Get(key)
+}
+
+// dbIndex returns the IST action index in the state database.
+func dbIndex(height uint32) []byte {
+	hBytes := make([]byte, 4)
+	binary.LittleEndian.PutUint32(hBytes, height)
+	return append([]byte(dbPrefix), hBytes...)
```

Comment:
Or, since Go 1.19, you can now do: `return binary.LittleEndian.AppendUint32([]byte(dbPrefix), height)` :)